File: SNS.Web/src/app/jobs/jobs/contracts/job-summary.dto.ts
import { JobType } from '../../enums/job-type.enum';
import { SalaryType } from '../../enums/salary-type.enum';

export interface JobSummaryDto {
    id: string;
    title: string;
    description?: string;
    companyId?: string;
    companyName: string;
    companyLogoUrl?: string | null;
    location: string;
    type: JobType;
    minSalary: number;
    maxSalary: number;
    currencyCode: string;
    salaryType: SalaryType;
    createdAt: string;
    closedAt?: string | null;
    isClosed?: boolean;
    isActive?: boolean;
    applicationsCount?: number;
    skills?: string[];
}

File: backend/SNS.Application/Jobs/Jobs/Contracts/JobSummaryDto.cs
using SNS.Domain.Jobs.Enums;

namespace SNS.Application.Jobs.Jobs.Contracts;

public sealed record JobSummaryDto(
    Guid Id,
    Guid CompanyId,
    string CompanyName,
    string? CompanyLogoUrl,
    string Title,
    string Description,
    string Location,
    JobType Type,
    decimal MinSalary,
    decimal MaxSalary,
    string CurrencyCode,
    SalaryType SalaryType,
    int ApplicationsCount,
    List<string> Skills,
    DateTime CreatedAt,
    DateTime? ClosedAt,
    bool IsActive
);
